### ISSUE TITLE
Doesn't report on warnings

### ISSUE THREAD

**Issue by vldslv-a** [ataylorme/eslint-annotate-action#80]
Hello! I've run into the following problem: it doesn't report eslint warnings
My config:

      - name: Annotate Code Linting Results
        uses: ataylorme/eslint-annotate-action@v3
        with:
          GITHUB_TOKEN: ${{ steps.generate-token.outputs.token }}
          report-json: reports/eslint_report.json
          only-pr-files: true
          fail-on-warning: true
          fail-on-error: true
          markdown-report-on-step-summary: true

As a result, I receive the following message:
<img width="718" alt="image" src="https://github.com/ataylorme/eslint-annotate-action/assets/97732940/dd6e9a51-f48b-455f-b84f-262985f5f555">
But I don't see  annotation about it in the PR. Could you fix it? This works well with errors, but not with warnings.

**Comment by beanow-at-crabnebula** [comment]
Likewise, if there are both errors and warnings, the errors get reported in the summary but the warnings do not.

Mentioning:
> Pull Request Changed Files ESLint Results:
>
> 1 ESLint error(s) and 0 ESLint warning(s) found

While this should actually be 1 error, 1 warning for my test case.

**Comment by beanow-at-crabnebula** [comment]
Disregard my comment.
The text actually gave the hint here. "**Pull Request Changed** Files".

Somewhere between `v1.5.1` and `v2.0.0` the behavior seems to have changed to only include errors/warning from code that was changed in that PR.

Maybe you were (also) expecting warnings from code that wasn't change in that PR?

**Comment by vldslv-a** [comment]
> Maybe you were (also) expecting warnings from code that wasn't change in that PR?

No, the problem is in the modified file.
A little above, I use `tj-actions/changed-files@v42` to get the changed files and run the linter only for them
With `only-pr-files: true`, I had problems that it took into account ALL files, not just the modified ones. But that's a different issue, I think)

**Comment by ataylorme** [comment]
Thanks for the feedback @vldslv-a! This is fixed in [v4.0.0-beta.1](https://github.com/ataylorme/eslint-annotate-action/releases/tag/v4.0.0-beta.1) — warnings now correctly produce annotations with `annotation_level: "warning"` instead of `"failure"`.

The new version also adds a `neutral-on-warning` input: when set to `true`, the check conclusion will be `neutral` (instead of `success`) when there are only warnings and no errors.

Please test the beta:
```yaml
- uses: ataylorme/eslint-annotate-action@v4.0.0-beta.1
  with:
    github-token: ${{ secrets.GITHUB_TOKEN }}
    report-json: eslint_report.json
    neutral-on-warning: true  # optional: neutral conclusion for warnings-only
```
